### DISCUSSION TITLE
No parse API exposed from @babel/standalone

### DISCUSSION THREAD

**Discussion by liuliangsir** [babel/babel#12482] [Q&A] [answered]
As the above title says, It seems that `@babel/standalone` only has `transform`,  `transformFromAst` and `transformScriptTags` API. That is to say, unlike `@babel/parser`, `@babel/standalone` has no `parse` API. What should I do when using `@babel/standalone` with `allowAwaitOutsideFunction` option? Last but not least, The top level await plugin is not suitable for me.

**Comment by JLHwung** [comment, marked as answer]
> @babel/standalone has no parse API.

`@babel/standalone` is focused on usage on Non-node.js environments. Personally I am not a fans of `parse` API in `@babel/standalone`. If you need to parse code in browsers (like astexplorer), you can use `@babel/parser` bundled with webpack or Rollup.

> What should I do when using @babel/standalone with allowAwaitOutsideFunction option?

https://babeljs.io/docs/en/options#parseropts

```js
transform(code, { parserOpts: { allowAwaitOutsideFunction: true } })
```

However it is unrecommended to use `allowAwaitOutsideFunction` as top level await should replace this option.
